pkg/storage/synchronization/lock_manager/sync_server.go

```diff
@@ -0,0 +1,179 @@
+	server, err := getSyncServerFromStorage(ctx, projectName, stagesStorage)
+	if err != nil {
+		return "", err
+	}
+
+	if server != "" {
+		logboek.Context(ctx).Debug().LogF("getSyncServerFromStorage %s selected server: %s\n", projectName, server)
+		return server, nil
+	}
+
+	return "", ErrNoSyncServerFound
+}
+
+// Create sync server name from repository or try to create record is doesn't exist
+func CreateSyncServerRecord(ctx context.Context, projectName, serverAddress string, stagesStorage storage.StagesStorage) error {
+	now := time.Now()
+	timestampMillisec := now.Unix()*1000 + now.UnixNano()/1000_000
+	rec := &storage.SyncServerRecord{Server: serverAddress, TimestampMillisec: timestampMillisec}
+
+	_, err := getSyncServer(ctx, projectName, stagesStorage)
+	if err != nil {
+		if errors.Is(err, ErrNoSyncServerFound) {
+			logboek.Context(ctx).Debug().LogF("СreateSyncServerRecord no syncserver found. Creating: %s\n", serverAddress)
+			if err := stagesStorage.PostSyncServerRecord(ctx, projectName, rec); err != nil {
+				return err
+			}
+			return nil
+		}
+		return err
+	}
+
+	logboek.Context(ctx).Info().LogF(fmt.Sprintf("СreateSyncServerRecord server already exists. %s will be used\n", serverAddress))
+
+	return nil
+}
+
+// Overwrite sync existed sync server
+func OverwriteSyncServerRepo(ctx context.Context, projectName, serverAddress string, stagesStorage storage.StagesStorage) error {
+	now := time.Now()
+	timestampMillisec := now.Unix()*1000 + now.UnixNano()/1000_000
+	rec := &storage.SyncServerRecord{Server: serverAddress, TimestampMillisec: timestampMillisec}
+
+	logboek.Context(ctx).Debug().LogF("СreateSyncServerRecord no synchronization server found. Creating: %s\n", serverAddress)
+	if err := stagesStorage.PostSyncServerRecord(ctx, projectName, rec); err != nil {
+		return fmt.Errorf("unable to overwrite sync server: %w", err)
+	}
+
+	return nil
+}
+
+func getSyncServerFromStorage(ctx context.Context, projectName string, stagesStorage storage.StagesStorage) (string, error) {
+	syncServerReords, err := stagesStorage.GetSyncServerRecords(ctx, projectName)
+	if err != nil {
+		return "", fmt.Errorf("can't get synchronization server records: %w", err)
+	}
+
+	if len(syncServerReords) > 0 {
+		res := selectOldestSyncServerRecord(syncServerReords)
+		logboek.Context(ctx).Debug().LogF("GetSyncServerFromStorage %s selected server: %s\n", projectName, res.Server)
+		return res.Server, nil
+	}
+
+	return "", nil
+}
+
+func selectOldestSyncServerRecord(records []*storage.SyncServerRecord) *storage.SyncServerRecord {
+	var foundRec *storage.SyncServerRecord
+	for _, rec := range records {
+		if foundRec == nil || rec.TimestampMillisec < foundRec.TimestampMillisec {
+			foundRec = rec
+		}
+	}
+	return foundRec
+}
+
+// It shows prompt message. Could be used if sync server differs from specified
+func PromptRewriteSyncRepoServer(ctx context.Context, specified, repoServer string) error {
```

Comment:
Warning before prompt would be nice.

`WARNING: A different sync server was detected than the one registered in the repository. This may cause inconsistencies in image builds. Ensure consistency or force the new one!`